misk/src/main/kotlin/misk/clustering/ClusterResourceMapper.kt

```diff
@@ -0,0 +1,13 @@
+package misk.clustering
+
+/**
+ * A [ClusterResourceMapper] maps string based resource IDs onto members of a cluster for the
+ * purposes of resource ownership. The default [ClusterResourceMapper] is a [ClusterHashRing]
+ * which performs a consistent hash across the cluster member, but [Cluster]s can supply their
+ * own mappings if there is a mechanism specific to that cluster (or to supply a fake)
+ */
+interface ClusterResourceMapper {
+  /** @return The [Cluster.Member] that should own the given resource id */
+  fun mapResourceToMember(resourceId: String): Cluster.Member
```

Comment:
maybe `operator fun get` ? That lets me do this:

```
val mapper: ClusterResourceMapper = ...
val memberForFoo = mapper["foo"]
```

Reply:
ah yeah i like that